### PR TITLE
ETH-675 Kick out a reverting mock-operator

### PR DESCRIPTION
**Pull request by samigabor**

### CODE REVIEW

File: packages/network-contracts/test/hardhat/OperatorTokenomics/Sponsorship.test.ts [hunk 1]
```diff
@@ -54,6 +54,34 @@ describe("Sponsorship contract", (): void => {
...
+        const { token } = contracts
+        const sponsorship = await deploySponsorshipWithoutFactory(contracts)
+        const badOperator = await (await getContractFactory("TestBadOperator", admin)).deploy()
+        await (await token.transfer(badOperator.address, parseEther("100"))).wait()
+        await (await badOperator.stake(sponsorship.address, parseEther("100"), token.address)).wait()
+
+        const sponsorshipBalanceBeforeUnstake = await token.balanceOf(sponsorship.address)
+        const operatorBalanceBeforeUnstake = await token.balanceOf(badOperator.address)
+        const sponsorshipStakeBeforeUnstake = await badOperator.getMyStake(sponsorship.address)
+        await (await badOperator.unstake(sponsorship.address)).wait()
```
Comment (jtakalai) [inline packages/network-contracts/test/hardhat/OperatorTokenomics/Sponsorship.test.ts:67]: ok this is a voluntary unstake, and I guess it still would work the same way: if your transferAndCall reverts, you just lose the tokens.

But the important thing to test is: sponsorship's IKickPolicy removes the operator. That they get removed.

File: packages/network-contracts/test/hardhat/OperatorTokenomics/Sponsorship.test.ts [hunk 2]
```diff
@@ -54,6 +54,34 @@ describe("Sponsorship contract", (): void => {
...
+        expect(sponsorshipStakeBeforeUnstake).to.equal(parseEther("100"))
+        expect(sponsorshipStakeAfterUnstake).to.equal(parseEther("0"))
+        // TestBadOperator.onTokenTransfer reverts so the funds did not get transferred
+        expect(operatorBalanceBeforeUnstake).to.equal(parseEther("0"))
+        expect(operatorBalanceAfterUnstake).to.equal(parseEther("0"))
+        expect(sponsorshipBalanceBeforeUnstake).to.equal(parseEther("100"))
+        expect(sponsorshipBalanceAfterUnstake).to.equal(parseEther("100"))
+
+        // bad operator got kicked out of the sponsorship even though he did not receive the funds from the previous unstake
+        await expect(badOperator.unstake(sponsorship.address))
```
Comment (jtakalai) [inline packages/network-contracts/test/hardhat/OperatorTokenomics/Sponsorship.test.ts:81]: you can test this more directly by asking `operator.stakedWei`

**Review by jtakalai — CHANGES_REQUESTED** [review]

File: packages/network-contracts/test/hardhat/OperatorTokenomics/Sponsorship.test.ts (hunk 1 above)
Comment (samigabor) [reply packages/network-contracts/test/hardhat/OperatorTokenomics/Sponsorship.test.ts:67]: Added another test where the bad operator is kicked through the admin policy

File: packages/network-contracts/test/hardhat/OperatorTokenomics/Sponsorship.test.ts (hunk 2 above)
Comment (samigabor) [reply packages/network-contracts/test/hardhat/OperatorTokenomics/Sponsorship.test.ts:81]: Updated

**Review by jtakalai — APPROVED** [review]

**Pull request merged**